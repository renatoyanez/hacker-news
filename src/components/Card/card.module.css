.card {
	border: solid 1px #979797;
	border-radius: 6px;
	overflow: hidden;
	height: 5.625rem;
	min-width: 34.375rem;
	max-width: 35rem;
	margin: 0 0 1.875em 0;
	padding: 0 0 0 1.625rem;
	flex: 0 1 calc(50% - 1em);
}

.card:hover {
	opacity: 0.5;
}

.cardContent {
	display: flex;
	justify-content: space-between;
	height: 100%;
}

.likeSection {
	display: flex;
	align-items: center;
	justify-content: center;
	min-width: 4.25rem;
	background-color: rgba(96, 96, 96, 0.06);
}

.infoSection {
	display: flex;
	flex-direction: column;
	justify-content: center;
}

@media screen and (max-width: 1100px) {
	.card {
		min-width: 15rem;
	}
}

@media screen and (max-width: 800px) {
	.card {
		min-width: 20rem;
		flex: 0 1 100%;
	}
}

@media screen and (max-width: 400px) {
	.card {
		max-width: 10rem;
		flex: 0 1 100%;
	}
}
