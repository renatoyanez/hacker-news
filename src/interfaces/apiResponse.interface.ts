export interface IApiResponse<T> {
	data: IResponseData<T>;
}

export interface IResponseData<T> {
	hits: T;
	page: number;
	nbPages: number;
	hitsPerPage: number;
	query: string;
}
