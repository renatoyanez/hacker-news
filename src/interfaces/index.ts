export * from './apiError.interface';
export * from './news.interface';
export * from './apiResponse.interface';
