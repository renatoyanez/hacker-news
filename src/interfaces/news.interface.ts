import { IApiResponse } from './apiResponse.interface';

export interface IHits {
	objectID: string;
	created_at: string;
	author: string;
	story_title: string;
	story_url: string;
}

export type INews = IApiResponse<IHits[]>;
