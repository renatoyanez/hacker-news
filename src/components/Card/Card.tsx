import { useState } from 'react';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
dayjs.extend(relativeTime);
import { IHits } from 'interfaces';
import { Like, Clock } from 'components/Icons';
import { Typography } from 'components';
import { useFaveNewsContext } from 'context/favesContext';
import styles from './card.module.css';

interface CardProps {
	hit: IHits;
	liked?: boolean;
	// handleFavedHits: (hit: IHits) => void;
	// storyTitle: string;
	// storyUrl: string;
	// date: string;
	// author: string;
	// faved?: boolean;
}

// const LikeSection = () => {
// 	return (
// 		<div className={styles.likeSection}>
// 			<Like />
// 		</div>
// 	);
// };

const Card = ({
	// storyTitle,
	// date,
	// author,
	// storyUrl,
	hit,
	liked,
}: // handleFavedHits,
CardProps) => {
	const { handleFavedHits, favedHits } = useFaveNewsContext();

	const [fromDate] = useState(dayjs().to(dayjs(hit.created_at)));
	const topText = `${fromDate} by ${hit.author}`;

	return (
		<article className={styles.card}>
			<div className={styles.cardContent}>
				<div className={styles.infoSection}>
					<a
						href={hit.story_url}
						target="_blank"
						style={{ textDecoration: 'none' }}
					>
						<div
							style={{
								display: 'flex',
								alignItems: 'center',
								// marginBottom: '0.44rem',
							}}
						>
							<div style={{ margin: '0.1rem 0.5rem 0 0' }}>
								<Clock />
							</div>
							<Typography type="textStyle6">{topText}</Typography>
						</div>
						<Typography type="textStyle3">
							{hit.story_title}
						</Typography>
					</a>
				</div>

				<div
					onClick={() =>
						handleFavedHits(hit, favedHits || ([] as IHits[]))
					}
					className={styles.likeSection}
				>
					<Like filled={liked} />
				</div>
			</div>
		</article>
	);
};

export default Card;
