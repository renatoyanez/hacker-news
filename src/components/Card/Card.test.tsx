import { ReactNode } from 'react';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';
import { IHits } from 'interfaces';
import Card, { CardProps } from './Card';

const hit: IHits = {
	objectID: '123',
	created_at: '1995-01-01',
	author: 'Author',
	story_title: 'the title',
	story_url: 'https://www.google.com/',
};

const basicProps: CardProps = {
	hit: {} as IHits,
	liked: false,
};

const props: CardProps = {
	hit: hit,
	liked: false,
};

describe('Card', () => {
	it('renders correctly', () => {
		const tree = create(<Card {...basicProps} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders with props', async () => {
		const { getByText } = render(<Card {...props} />);

		const placeholder = getByText(props.hit.story_title);
		expect(placeholder).toBeTruthy();
	});
});
