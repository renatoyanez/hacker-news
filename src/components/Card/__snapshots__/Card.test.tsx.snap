// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<article
  className="card"
>
  <div
    className="cardContent"
  >
    <div
      className="infoSection"
    >
      <a
        style={
          Object {
            "textDecoration": "none",
          }
        }
        target="_blank"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0.1rem 0.5rem 0 0",
              }
            }
          >
            <svg
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 1.333c3.676 0 6.667 2.991 6.667 6.667 0 3.676-2.991 6.667-6.667 6.667-3.676 0-6.667-2.991-6.667-6.667 0-3.676 2.991-6.667 6.667-6.667zM8 0C3.582 0 0 3.582 0 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zm.667 8V4H7.333v5.333H12V8H8.667z"
                fill="#606060"
              />
            </svg>
          </div>
          <p
            className="textStyle6"
          >
            a few seconds ago by undefined
          </p>
        </div>
        <p
          className="textStyle3"
        />
      </a>
    </div>
    <div
      className="likeSection"
      data-testid="like-icon"
      onClick={[Function]}
    >
      <svg
        height="22"
        viewBox="0 0 24 22"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 8.229C12.234 7.109 13.547 2 17.382 2 19.602 2 22 3.551 22 7.003c0 3.907-3.627 8.47-10 12.629C5.627 15.473 2 10.91 2 7.003c0-3.484 2.369-5.005 4.577-5.005 3.923 0 5.145 5.126 5.423 6.231zM0 7.003C0 11.071 3.06 16.484 12 22c8.94-5.516 12-10.929 12-14.997 0-7.962-9.648-9.028-12-3.737C9.662-1.996 0-1.004 0 7.003z"
          fill="#DD0031"
        />
      </svg>
    </div>
  </div>
</article>
`;
